### DISCUSSION TITLE
MQTT topics incomplete for most clients

### DISCUSSION THREAD

**Discussion by electric-m** [Koenkk/zigbee2mqtt#10891] [General]
**My setup (simplified):**

Ubuntu VM running on Proxmox
ioBroker MQTT Client-Adapter running nativley under the Ubuntu VM
Mosquitto Broker running in Docker on Ubuntu VM
Zigbee2MQTT running in Docker on Ubuntu VM
1 Sensor Temp and Humidity Xiaomi WSDCGQ01LM

**Problem:**

Only ioBroker MQTT Client receives the full sensor values:
<img width="1718" alt="Bildschirmfoto 2022-01-19 um 16 39 13" src="https://user-images.githubusercontent.com/40071240/150165748-b5dff949-67f2-4756-b4a9-6694af0f31e9.png">

Other MQTT Clients (Example here with MQTT Explorer only receive the topic "availability" when subscribing to either zigbee2mqtt/# or even # .

The sensor-topics like temperature and so on are not received, or at least I cannot find them.
Tried several clients.

Any help much appreciated.

<img width="1136" alt="Bildschirmfoto 2022-01-19 um 16 38 04" src="https://user-images.githubusercontent.com/40071240/150166129-b798b3ed-ae89-4851-8fd4-cd349bf7a6a9.png">

**Comment by electric-m** [comment]
For now, I fixed it by reflashing the Zigbee Stick with ZigStar GW Tool and latest Router firmware.
Will have to see if that is table now.

**Comment by electric-m** [comment]
As I wrote above, I had all the sensor exposes as topics availabe shortly.

Now they are gone again, only availabilty is there again.

**Reply by Koenkk** [reply]
Aavailability is published as retained, the rest not so you have to wait until it is published again OR enable retained for all devices by adding to `configuration.yaml`.

```yaml
device_options:
  retain: true
```

**Reply by electric-m** [reply]
Thanks I just figured it out and set retain = true through web IF.
